pandas/io/common.py

```diff
@@ -836,19 +852,30 @@ def __getattr__(self, name: str):
     def __iter__(self) -> _MMapWrapper:
         return self
 
+    def read(self, size: int = -1) -> str | bytes:
```

Comment:
This function could be removed if the c-engine handled non-utf-8 better.

The PR in its current form will in case of the c-engine: 1) use mmap to read the entire file 2) decode it appropriately (this function) 3) the c-code will encode the now utf-8 string into bytes again. It would be more efficient if the c-engine supported non-utf-8 in more places. I will look into that, but that might take some time.

Reply:
prob not worth the effort to handle non-utf-8 better, but if you want to look...ok